pkg/services/librarypanels/database.go

```diff
@@ -113,13 +113,20 @@ func (lps *LibraryPanelService) connectLibraryPanelsForDashboard(c *models.ReqCo
 
 // deleteLibraryPanel deletes a Library Panel.
 func (lps *LibraryPanelService) deleteLibraryPanel(c *models.ReqContext, uid string) error {
-	orgID := c.SignedInUser.OrgId
 	return lps.SQLStore.WithTransactionalDbSession(context.Background(), func(session *sqlstore.DBSession) error {
-		result, err := session.Exec("DELETE FROM library_panel WHERE uid=? and org_id=?", uid, orgID)
+		panel, err := getLibraryPanel(session, uid, c.SignedInUser.OrgId)
 		if err != nil {
 			return err
 		}
 
+		if _, err := session.Exec("DELETE FROM library_panel_dashboard WHERE librarypanel_id=?", panel.ID); err != nil {
+			return err
+		}
+
+		result, err := session.Exec("DELETE FROM library_panel WHERE id=?", panel.ID)
+		if err != nil {
+			return err
+		}
```

Comment:
I went back and forth on the order for this. Should we delete the connections before the panel or the other way around?

Reply:
Connections first, but it's all in a transaction anyway isn't it, so shouldn't matter to the outside world.

Reply:
Yeah, I finally got it right 😉